### PR TITLE
Update dist.scatter() documentation

### LINKED ISSUE #84566: Update dist.scatter() documentation to provide clearer arg description
### 📚 The doc issue

https://pytorch.org/docs/master/distributed.html#torch.distributed.scatter documentation does not specify that scatter_list tensors must have same sizes.

Example:

```python
import torch
import torch.distributed as dist
import os
import torch.multiprocessing as mp

os.environ["MASTER_ADDR"] = "localhost"
os.environ["MASTER_PORT"] = "29500"
os.environ["TORCH_SHOW_CPP_STACKTRACES"] = "1"

def work(rank, world_size):
    dist.init_process_group("gloo", rank=rank, world_size=world_size)
    print("finished creating process group")
    t_13 = torch.zeros(13)
    t_6 = torch.zeros(6)
    output_tensor = torch.zeros(13)
    if rank == 0:
        dist.scatter(output_tensor, [t_13, t_6])
    else:
        dist.scatter(output_tensor)
    print(f"rank {rank} - {output_tensor=}")

if __name__ == "__main__":
    world_size = 2
    mp.spawn(work, nprocs=world_size, args=(world_size,))
```

error:

```
RuntimeError: ProcessGroupGloo::scatter: invalid tensor size at index 1 (expected (13), got (6))
```

### Suggest a potential alternative/fix

We should also include an example usage of scatter in the documentation.

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @kwen2501

### PR DESCRIPTION
**Pull request by zerolinux5**
Update documentation for dist. scatter

Fixes #84566

### CODE REVIEW

File: torch/distributed/distributed_c10d.py [hunk 1]
```diff
@@ -2556,6 +2556,26 @@ def scatter(tensor, scatter_list=None, src=0, group=None, async_op=False):
...
+        >>> # xdoctest: +SKIP("need process group init")
+        >>> # Note: Process group initialization omitted on each rank.
+        >>> import torch.distributed as dist
+        >>> size_of_tensor = 2
+        >>> t_ones = torch.ones(size_of_tensor)
+        >>> t_fives = torch.ones(size_of_tensor) * 5
+        >>> output_tensor = torch.zeros(size_of_tensor)
+        >>> if dist.get_rank() == 0:
+        >>>     # Assumes world_size of 2.
+        >>>     scatter_list = [t_ones, t_fives] # Only Tensors
```
Comment (rohan-varma) [inline torch/distributed/distributed_c10d.py:2571]: nit: only tensors, all of which must be the same size.

**Review by rohan-varma — APPROVED** [review]
LGTM, @H-Huang does this address all the doc improvements? 

Also, could you add a clearer PR title?

File: torch/distributed/distributed_c10d.py [hunk 2]
```diff
@@ -2556,6 +2556,26 @@ def scatter(tensor, scatter_list=None, src=0, group=None, async_op=False):
         Async work handle, if async_op is set to True.
         None, if not async_op or if not part of the group
 
+    .. note:: Note that all Tensors in scatter_list must have the same size.
+
+    Example::
+        >>> # xdoctest: +SKIP("need process group init")
+        >>> # Note: Process group initialization omitted on each rank.
+        >>> import torch.distributed as dist
+        >>> size_of_tensor = 2
```
Comment (H-Huang) [inline torch/distributed/distributed_c10d.py:2565]: nit:
```suggestion
        >>> tensor_size = 2
```

File: torch/distributed/distributed_c10d.py [hunk 3]
```diff
@@ -2556,6 +2556,26 @@ def scatter(tensor, scatter_list=None, src=0, group=None, async_op=False):
...
+        >>> output_tensor = torch.zeros(size_of_tensor)
+        >>> if dist.get_rank() == 0:
+        >>>     # Assumes world_size of 2.
+        >>>     scatter_list = [t_ones, t_fives] # Only Tensors
+        >>> else:
+        >>>     scatter_list = None
+        >>> dist.scatter(output_tensor, scatter_list, src=0)
+        >>> # Rank i gets scatter_list[i]. For example, on rank 1:
+        >>> output_tensor
+        tensor([5., 5., 5.])
```
Comment (H-Huang) [inline torch/distributed/distributed_c10d.py:2577]: your tensor size is two, so there should only be two 5s

**Review by H-Huang — COMMENTED** [review]
Thanks! left some comments

**Review by H-Huang — APPROVED** [review]
LGTM, thanks again for adding!

**Pull request closed without merging**